src/librbd/ImageCtx.cc

```diff
@@ -849,8 +849,11 @@ struct C_InvalidateCache : public Context {
     // ensure no locks are held when flush is complete
     on_safe = util::create_async_context_callback(*this, on_safe);
 
+    bool journaling = false;
+    journaling = (journal != NULL && !journal->is_journal_replaying());
+
```

Comment:
Technically it probably makes sense to move the assertion up -- it will potentially crash either way if called w/o holding the owner lock.